bin/initoverlayfs-install

```diff
@@ -1,58 +1,105 @@
 #!/bin/bash
 
-set -ex
+set -e
 
-extract_initrd_into_initoverlayfs() {
-  DIR_TO_DUMP_INITRAMFS="/var/tmp/initoverlayfs"
-  mkdir -p "$DIR_TO_DUMP_INITRAMFS"
-
-  file_type=$(file /boot/initramfs-$kver.img)
-  decompressor="lz4cat"
-  decompressor_dracut="--lz4"
-  if [[ "$file_type" == *"ASCII cpio archive (SVR4 with no CRC)"* ]]; then
-    decompressor_dracut=""
-    decompressor="zcat"
-  elif [[ "$file_type" == *"regular file, no read permission"* ]] || \
-       [[ "$file_type" == *"gzip"* ]]; then
-    decompressor_dracut=""
-    decompressor="zcat"
-  fi
-
-  if command -v mkfs.erofs; then
-    fstype="erofs"
-    cd "$DIR_TO_DUMP_INITRAMFS"
-    /usr/lib/dracut/skipcpio /boot/initramfs-$kver.img | $decompressor | cpio -ivd
-    cd -
-    rm -f /boot/initoverlayfs-$kver.img
-    mkfs.erofs /boot/initoverlayfs-$kver.img "$DIR_TO_DUMP_INITRAMFS"
-  else
-    echo "Please install mkfs.erofs"
-    exit 0
-
-    fstype="ext4"
-    dd if=/dev/zero of=/boot/initoverlayfs-$kver.img bs=64M count=1
-    dev=$(losetup -fP --show /boot/initoverlayfs-$kver.img)
-    mkfs.$fstype $dev
-    mount $dev "$DIR_TO_DUMP_INITRAMFS"
-    cd "$DIR_TO_DUMP_INITRAMFS"
-    /usr/lib/dracut/skipcpio /boot/initramfs-$kver.img | zstd -d --stdout | cpio -ivd
-    sync
-    cd -
-    while ! umount "$DIR_TO_DUMP_INITRAMFS"; do
+# Constants with default value
+DECOMPRESSOR="lz4cat"
+DECOMPRESSOR_DRACUT="--lz4"
+
+INITOVERLAYFS_CONF="/etc/initoverlayfs.conf"
+INITRAMFS_DUMP_DIR="/var/tmp/initoverlayfs"
+
+SUPPORTED_FILESYSTEM=("erofs" "ext4")
+SKIPCPIO_BIN="/usr/lib/dracut/skipcpio"
+
+exec_erofs() {
+    pushd "${INITRAMFS_DUMP_DIR}" || exit
+        "${SKIPCPIO_BIN}" /boot/initramfs-"${kver}".img | "${DECOMPRESSOR}" | cpio -ivd
+    popd || exit
+    rm -f /boot/initoverlayfs-"${kver}".img
+    mkfs.erofs /boot/initoverlayfs-"${kver}".img ${INITRAMFS_DUMP_DIR}
+}
+
+exec_ext4() {
```

Comment:
I'm don't think this actually works, that's why the `exit 0` was there, so maybe we could do something like

```
echo "Only erofs-based initoverlayfs supported, ext4, btrfs, xfs, etc. to be implemented"
exit 0
```